app/common/util/src/main/java/io/syndesis/common/util/Json.java

```diff
@@ -85,6 +97,39 @@ public static ObjectMapper copyObjectMapperConfiguration() {
         return OBJECT_MAPPER.copy();
     }
 
+    /**
+     * This method creates a copy of the default ObjectMapper configuration and adds special Json schema compatibility handlers
+     * for supporting draft-03, draft-04 and draft-06 level at the same time.
+     *
+     * Auto converts "$id" to "id" property for draft-04 compatibility.
+     *
+     * In case the provided schema specification to read uses draft-04 and draft-06 specific features such as "examples" or a list of "required"
+     * properties as array these information is more or less lost and auto converted to draft-03 compatible defaults. This way we can
+     * read the specification to draft-03 compatible objects and use those.
+     * @return
+     */
+    public static ObjectReader defaultJsonSchemaReader() {
```

Comment:
Naming this method like this it's very likely for someone to use it without realizing it deals with specific issue in parsing JSON schemas (think someone just auto completing `defaultJson...`). In general this functionality is not _common_ so I'd move it somewhere else. Preferably out of the common module.

Reply:
I can change the name to not use `default`. Something like `jsonSchemaReader()`

Though we need to keep it in some common module as it is being used not only by schema inspector but also within split/aggregate step handlers and split/aggregate meta data handlers.

Reply:
it is quite important to also use the other default object mapper settings as basis of this Json schema reader instance. We need the enable/disable settings done here: https://github.com/syndesisio/syndesis/blob/ef2d1189b3ac8e183d32f781a8f58a01b419becd/app/common/util/src/main/java/io/syndesis/common/util/Json.java#L55

Reply:
Perhaps a separate class for JSON Schema, like we have one for OpenAPI stuff. I think it should concern us to put more stuff into common, it's starting to smell...

Reply:
So where would I put this new class? into a separate module like https://github.com/syndesisio/syndesis/tree/master/app/server/api-generator?

Reply:
If it needs to be in common i.e. a lot of modules depend on it, so that makes it easier, then perhaps a separate class in a separate package like we have for OpenAPI: https://github.com/syndesisio/syndesis/tree/master/app/common/util/src/main/java/io/syndesis/common/util/openapi

Reply:
so if this all stays in common module I think we should use https://github.com/syndesisio/syndesis/tree/master/app/common/util/src/main/java/io/syndesis/common/util/json and move everything related to Json utils to this package then. I would like to do this in a separate PR as this is not related to the initial bug that we try to fix here.

Reply:
Well, we're discussing the placement of code added in this PR, so not sure about a separate PR to follow up on that. I'd prefer a separate class as anything regarding JSON schema should be optional when depending on `common-util` and putting it in the Json class makes it mandatory. I'd prefer a separate package as any class (the mixin for example) can be potentially made package private as we would not want to expose it as an API that might get depended on.